airbyte-cdk/python/airbyte_cdk/sources/file_based/file_based_source.py

```diff
@@ -29,12 +31,14 @@ def __init__(
         stream_reader: AbstractFileBasedStreamReader,
         catalog: Optional[ConfiguredAirbyteCatalog],
         availability_strategy: Optional[AvailabilityStrategy],
+        spec_class: Type[AbstractFileBasedSpec],
         discovery_policy: AbstractDiscoveryPolicy = DefaultDiscoveryPolicy(),
         parsers: Dict[str, FileTypeParser] = None,
         validation_policies: Dict[str, AbstractSchemaValidationPolicy] = DEFAULT_SCHEMA_VALIDATION_POLICIES,
     ):
         self.stream_reader = stream_reader
         self.availability_strategy = availability_strategy or DefaultFileBasedAvailabilityStrategy(stream_reader)
+        self.spec_class = spec_class
```

Comment:
i made this required, there _could_ be a case for this spec object being optional, but given that every source should at least verify a documentation url and will almost always require at least one unique config option, my gut tells me this will always need to be implemented